model/custom_status.go

```diff
@@ -30,10 +30,11 @@ var validCustomStatusDuration = map[string]bool{
 }
 
 type CustomStatus struct {
-	Emoji     string    `json:"emoji"`
-	Text      string    `json:"text"`
-	Duration  string    `json:"duration"`
-	ExpiresAt time.Time `json:"expires_at"`
+	Emoji        string    `json:"emoji"`
+	Text         string    `json:"text"`
+	Duration     string    `json:"duration"`
+	ExpiresAt    time.Time `json:"expires_at"`
+	SetByProduct bool      `json:"set_by_product,omitempty"`
```

Comment:
Not sure if `SetByProduct` is descriptive, maybe focusing more on the behavior rather than in "who" set it? like `IsSetAutomatically` or something like this? also changing the name to reference it's a boolean (that's why I added `Is` at the beginning).